store/userinfo.go

```diff
@@ -51,47 +49,38 @@ type User struct {
 }
 
 func UserInfo(email string) (userinfo *User, err error) {
+	var v keysReply
 	ssourl := fmt.Sprintf("%s/keys/%s", authURL(), url.QueryEscape(email))
-	for attempt := retry.Start(defaultRetryStrategy, nil); attempt.Next(); {
-		var resp *http.Response
-		resp, err = httpClient.Get(ssourl)
-		if err != nil {
-			if shouldRetryError(attempt, err) {
-				continue
-			}
-			break
-		}
-
-		if shouldRetryHttpResponse(attempt, resp) {
-			resp.Body.Close()
-			continue
-		}
 
-		defer resp.Body.Close()
-
-		switch resp.StatusCode {
-		case 200: // good
-		case 404:
-			return nil, fmt.Errorf("cannot find user %q", email)
-		default:
-			return nil, respToError(resp, fmt.Sprintf("look up user %q", email))
+	resp, err := httputil.RetryRequest(ssourl, func() (*http.Response, error) {
+		return httpClient.Get(ssourl)
+	}, func(resp *http.Response) error {
+		if resp.StatusCode != 200 {
+			// we recheck the status
+			return nil
 		}
-
-		var v keysReply
 		dec := json.NewDecoder(resp.Body)
-		if err = dec.Decode(&v); err != nil {
-			if shouldRetryError(attempt, err) {
-				continue
-			}
-			return nil, fmt.Errorf("cannot unmarshal: %s", err)
+		if err := dec.Decode(&v); err != nil {
+			return fmt.Errorf("cannot unmarshal: %v", err)
 		}
+		return nil
+	}, defaultRetryStrategy)
+
+	if err != nil {
+		return nil, err
+	}
 
-		return &User{
-			Username:         v.Username,
-			SSHKeys:          v.SSHKeys,
-			OpenIDIdentifier: v.OpenIDIdentifier,
-		}, nil
+	switch resp.StatusCode {
+	case 200: // good
+	case 404:
```

Comment:
Also, I agree with @chipaca about dropping Context.. Not sure that's a good idea, perhaps it could be moved to requestOptionis?